Flaky test TestServerGracefulExit

```
=== RUN   TestServerGracefulExit
    server_test.go:677: 
        	Error Trace:	server_test.go:677
        	Error:      	Should be true
        	Test:       	TestServerGracefulExit
        	Messages:   	Error log found on server exit: {{error 2021-05-21 19:22:10.814114204 +0000 UTC m=+2.124395084  Could not start HTTP server undefined } [{error 26 0  mux: server closed}]}
--- FAIL: TestServerGracefulExit (1.00s)
```
Source: `./cmd/query/app/server_test.go:651:func TestServerGracefulExit(t *testing.T)`

--- a/cmd/query/app/server.go
+++ b/cmd/query/app/server.go
@@ -235,7 +235,7 @@ func (s *Server) Start() error {
 			err = s.httpServer.Serve(s.httpConn)
 		}
 		switch err {
-		case nil, http.ErrServerClosed, cmux.ErrListenerClosed:
+		case nil, http.ErrServerClosed, cmux.ErrListenerClosed, cmux.ErrServerClosed:
 			// normal exit, nothing to do
 		default:
 			s.logger.Error("Could not start HTTP server", zap.Error(err))

